### DISCUSSION TITLE
TEA Party Help Desk epoch7.1 design

### DISCUSSION THREAD

**Discussion by kevingzhang** [tearust/teaproject#109] [Ideas]
# Goal
- A paid Quora or Stackoverflow
- NPC issues credit to every new user as "tuition credit". They can use those credit to pay for questions here
- Existing veteran users can answer questions and "earn" the tuition if his answer is marked "correct & helpful"
- NPC can pin the long lasting questions. and give tip to both user who ask and answer
- Demonostrate what a TApp business should run without Web  2.0 thinking

# TEA Party vs Help Desk
TEA Party is a "App Generator". User can use TEA Party as a template to generate individual tapp. Help Desk is one of the TApps that generated using TEA party. In other words, Help desk is a "Tea table" of TEA Party there are many tea table in the tea party.
Another most wanted Tea Table is "Bullet board" that user can post ads (Seeds wanted, CML for sale etc).

# Basic rules
User A asks a question and posts it to the Help Desk. He will need to
- pay a deposit for both hosting fees and answer reward in total
- As long as the question stay alive on the board, the cost of hosting is charged based on time
- If someone answers his question, the answer will be listed below his question.
- Any community member can "upvote" an answer if he thinks the answer is the best. There is an upvote fee.
- Any community member can "downvote" an answer if he thinks the answer is incorrect or misleading. this is a downvote fee.
- The original asker can mark an answer "Correct".  
- Selected "Correct" answerer will get paid the tip
- If someone "upvote" "Correct" an answer, he will get paid as well by his "upvote"
- If someone "downvote" a "Correct" answer, he will not get rewarded and his "down vote" fee is sent to the "correct answer"
- If a "resolved" is marked. People can still upvote /downvote, but no more reward

# Motivation
If someone answer questions and marked correct. he will get major reward from the asker.
If someone upvote an answer, and finally marked as "correct", he also receive reward (from those who did not vote this answer)
if someone downvote an answer and finally this answer is marked correct. His "downvote" fee will be send to the "upvoter" to this question.
If someone upvote to an answer that is not marked as "correct" answer, his "upvote" will go to the "correct" answer pool

NPC will read all the question and answers, if a question and answer is worth to help most others, NPC will "pin" this thread. NPC will pay for the host. NPC also gives a tip to both the answerer and the asker.

**Comment by geopor** [comment]
#### Why Downvote?
- If upvoters choose correctly, they are rewarded by the pool funded by downvoters. But I don't understand the economic motivation to "downvote": it costs money and you're never rewarded for doing so.

#### Making Money
- If TApp creator wants to make money, they have to cover their cost: pay a deposit for both hosting fees and answer reward in total. They must cover this cost through the consume action. They can also save money by choosing their own answer as correct.
- For consumers using the TApp, even if reward is large, they may not trust TApp creator to select the correct answer even if their answer is best. So not sure if reward will motivate them. They can also upvote  and deposit some small fee for each upvote that will ultimately be split among the people who choose the correct answer. This could work well for a prediction market if they can trust in the "correct" answer being selected (for example, the winner of a horse race). For reward to motivate people, they have to have trust that it will go to right person. Prediction markets get around this by using oracles - oracle determines "correct" answer, not TApp creator.

**Comment by geopor** [comment]
We are now working on an offshoot of this idea here -> https://github.com/tearust/teaproject/discussions/118
